addon/utils/selection.js

```diff
@@ -63,24 +62,27 @@ export default {
    * @param {Function} itemComparator - comparator for items
    */
   /* eslint-disable complexity */
-  range (items, selectedItems, item, rangeState, itemComparator) {
+  range (items, selectedItems, item, rangeState, itemComparator, itemKey) {
     // If an anchor isn't set, then set the anchor and exit
     const rangeAnchor = rangeState['anchor']
-    if (isNone(rangeAnchor)) {
+
+    const anchor = items.findIndex(currentItem => itemComparator(currentItem, rangeState['anchor']))
+    // If anchor is -1 then it was a anchor from a previous page that we can not find, so reset
+    if (Ember.isEmpty(rangeAnchor) || anchor === -1) {
```

Comment:
isNone is not good with the new item key.
itemComparators  .get method does not work on null items.
isEmpty tests for empty ember objects based what I googled. 
It's also passing all the tests in expected behavior so I believe its working as intended.